### ISSUE TITLE
Spans are not enclosed by their background correctly

### ISSUE THREAD

**Issue by halfer** [dompdf/dompdf#690]
There's three (presumably related) issues I've identified here:
- Spans with a coloured background do not contain the descenders of the text within
- Padding on a span is not added to its bottom side
- Margins on a span affect its text but not its background

Here's a document to reproduce:

``` html
    <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
    <html xmlns="http://www.w3.org/1999/xhtml">
    <head>
      <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
      <meta http-equiv="Content-Style-Type" content="text/css" />
      <meta name="generator" content="pandoc" />
      <title>DOMPDF padding test</title>
      <style type="text/css">code{white-space: pre;}</style>
      <style type="text/css" >
          .about {
              color: grey;
              font-style: italic;
          }
          .highlight {
              background-color: greenyellow;
          }
          .highlight-padding {
              background-color: orange;
              padding: 6px;
          }
          .highlight-margin {
              margin: 100px 20px;
              background-color: red;
          }
      </style>
    </head>
    <body>
        <h1>Padding test</h1>

        <p>This bug probably affects all inline spans. I've illustrated <code>span</code> and <code>code</code>,
            since my live use case makes use of the latter.</p>

        <p class="about">1. Browsers render inline block backgrounds to include descenders, but dompdf does not:</p>

        <p>Here is <span class="highlight">a span</span> with no padding.</p>
        <p>Here is <code class="highlight">an inline code block (descender: q)</code> with no padding.</p>

        <p class="about">2. In a browser, padding is added to all sides of a span, but in dompdf it doesn't cover
            the descender and it is not added to the bottom:</p>

        <p>Here is <span class="highlight-padding">a span</span> with padding.</p>
        <p>Here is <code class="highlight-padding">an inline code block (descender: q)</code> with padding.</p>

        <p class="about">3. Browsers treat margins on inline spans as having no height, so top and bottom do not
            affect the line height. Left and right space outside the coloured box should be added though. In
            dompdf, the left and right margin is respected for the text but not its containing box:
        </p>

        <p>Here is <span class="highlight-padding highlight-margin">a span</span> with padding and margin.</p>  
        <p>Here is <code class="highlight-padding highlight-margin">an inline code block (descender: q)</code> with padding and margin.</p>
    </body>
    </html>
```

I'll add a screenshot for reference too. Thanks for your work on this - dompdf has already fixed some issues that wkhtmltopdf could not!

I'm on PHP 5.3.25 via Macports on Mac 10.6.8. Using latest Firefox as browser reference. Rendering using dompdf version 54551d324.

**Comment by roberthenniger** [comment]
@bsweeney yes the wrong width calculation with inline-block is really frustrating. Is there a fix planned for this?

**Comment by bsweeney** [comment]
Would love to produce a fix but can't say when that would happen.

**Comment by bsweeney** [comment]
Dompdf is much improved rendering this document, particularly with the upcoming release. Descenders, however, still aren't fully contained.

**Comment by bsweeney** [comment]
I believe this issue was sufficiently addressed with the 2.0.0 release. There appears to be a slight under-calculation (a pixel or two) when the core fonts are used, but I think we can close this issue out.
